Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssSecurity.ps1

```diff
@@ -0,0 +1,169 @@
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#>
+
+[DscResource()]
+class VMHostVssSecurity : VMHostVssBaseDSC {
+    <#
+    .DESCRIPTION
+
+    The flag to indicate whether or not all traffic is seen on the port.
+    #>
+    [DscProperty()]
+    [boolean] $AllowPromiscuous
+
+    <#
+    .DESCRIPTION
+
+    The flag to indicate whether or not the virtual network adapter should be
+    allowed to send network traffic with a different MAC address than that of
+    the virtual network adapter.
+    #>
+    [DscProperty()]
+    [boolean] $ForgedTransmits
+
+    <#
+    .DESCRIPTION
+
+    The flag to indicate whether or not the Media Access Control (MAC) address
+    can be changed.
+    #>
+    [DscProperty()]
+    [boolean] $MacChanges
+
+    [void] Set() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+
+        $this.UpdateVssSecurity($vmHost)
+    }
+
+    [bool] Test() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+        $vss = $this.GetVss($vmHost)
+
+        if ($this.Ensure -eq [Ensure]::Present) {
+            return ($null -ne $vss -and $this.Equals($vss))
+        }
+        else {
+            $this.AllowPromiscuous = $false
+            $this.ForgedTransmits = $true
+            $this.MacChanges = $true
+
+            return ($null -eq $vss -or $this.Equals($vss))
+        }
+    }
+
+    [VMHostVssSecurity] Get() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $result = [VMHostVssSecurity]::new()
+        $result.Server = $this.Server
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+
+        $result.Name = $vmHost.Name
+        $this.PopulateResult($vmHost, $result)
+
+        if ('' -ne $result.VssName) {
```

Comment:
The same refactoring as the VMHostVss Resource.

Reply:
Changed